### PR TITLE
Additional remotes

### PR DESCRIPTION
**Pull request by stouf**
# What is in ?

This Pull Request lets the developer specify its own remotes when calling the `component-proxy-install` binary.

See the issue https://github.com/Wizcorp/component-proxy-install/issues/2

### CODE REVIEW

File: bin/component-proxy-install [hunk 1]
```diff
@@ -7,8 +7,18 @@ console.log('Starting component proxy');
...
 
 var port = app.getPort();
+var remotes = 'https://raw.githubusercontent.com,http://127.0.0.1:' + port;
 
-var installArgs = [ 'install', '-r', 'https://raw.githubusercontent.com,http://127.0.0.1:' + port ];
+// Look for remote options in the passed arguments
+for (var i = 2; i < process.argv.length; i = i + 1) {
+	if (process.argv[i] === '-r' || process.argv[i] === '--remotes') {
+		remotes += (',' + process.argv[i + 1]);
+		process.argv.splice(i, 2);
```
Comment (ronkorving) [inline bin/component-proxy-install]: Please do not mess with process.argv. You may read and copy it, but please do not modify it.

File: bin/component-proxy-install [hunk 2]
```diff
@@ -7,8 +7,18 @@ console.log('Starting component proxy');
 app.start();
 
 var port = app.getPort();
+var remotes = 'https://raw.githubusercontent.com,http://127.0.0.1:' + port;
 
-var installArgs = [ 'install', '-r', 'https://raw.githubusercontent.com,http://127.0.0.1:' + port ];
+// Look for remote options in the passed arguments
+for (var i = 2; i < process.argv.length; i = i + 1) {
+	if (process.argv[i] === '-r' || process.argv[i] === '--remotes') {
+		remotes += (',' + process.argv[i + 1]);
```
Comment (ronkorving) [inline bin/component-proxy-install]: Why the braces?

Comment (stouf) [inline bin/component-proxy-install]: No particular reason.... Should I remove??

File: bin/component-proxy-install (hunk 1 above)
Comment (stouf) [inline bin/component-proxy-install]: Oh yeah, right !

File: bin/component-proxy-install [hunk 3]
```diff
@@ -7,8 +7,18 @@ console.log('Starting component proxy');
 app.start();
 
 var port = app.getPort();
+var remotes = 'https://raw.githubusercontent.com,http://127.0.0.1:' + port;
```
Comment (bjornstar) [inline bin/component-proxy-install]: Please make this an array.

File: bin/component-proxy-install (hunk 2 above)
Comment (ronkorving) [inline bin/component-proxy-install]: I would remove them :)

Comment (bjornstar) [inline bin/component-proxy-install]: Then you can push the remote on

File: bin/component-proxy-install [hunk 4]
```diff
@@ -7,8 +7,18 @@ console.log('Starting component proxy');
...
+// Look for remote options in the passed arguments
+for (var i = 2; i < process.argv.length; i = i + 1) {
+	if (process.argv[i] === '-r' || process.argv[i] === '--remotes') {
+		remotes += (',' + process.argv[i + 1]);
+		process.argv.splice(i, 2);
+		i = i - 1;
+	}
+}
+
+var installArgs = [ 'install', '-r', remotes];
```
Comment (bjornstar) [inline bin/component-proxy-install]: and finally, join the remotes.

**Comment by stouf** [comment]
Updated.

**Comment by ronkorving** [comment]
lgtm :shipit:

**Pull request merged**